src/yunohost/migrations.py

```diff
@@ -0,0 +1,249 @@
+
+            try:
+                if mode == "forward":
+                    migration["module"].MyMigration().migrate()
+                elif mode == "backward":
+                    migration["module"].MyMigration().backward()
+                else:  # can't happend
+                    raise Exception()
+            except Exception as e:
+                # migration failed, let's stop here but still update state because
+                # we managed to run the previous ones
+                logger.error("Migration {number} {name} has failed with exception {exception}, abording".format(exception=e, **migration), exc_info=1)
+                break
+
+        else:  # if fake
+            logger.warn("Fake migration {number} {name}...".format(**migration))
+
+        # update the state to include the latest runed migration
+        state["last_runed_migration"] = {
+            "number": migration["number"],
+            "name": migration["name"],
+        }
+
+    # special case where we want to go back from the start
+    if target == 0:
+        state["last_runed_migration"] = None
+
+    try:
+        state = json.dumps(state, indent=4)
+    except Exception as e:
+        raise MoulinetteError(
+            errno.EINVAL,
+                "Unable to serialize migration state ('{state}') in json because of exception {exception}".format(
+                exception=e,
+                state=state,
+            )
+        )
+
+    try:
+        open(MIGRATIONS_STATE_PATH, "w").write(state)
+    except Exception as e:
+        raise MoulinetteError(
+            errno.EINVAL,
+                "Unable to save migration state ('{state}') at {path} because of exception {exception}".format(
+                exception=e,
+                path=MIGRATIONS_STATE_PATH,
+                state=state,
+            )
+        )
+
+
+def migrations_state():
+    """
+    Show current migration state
+    """
+    if not os.path.exists(MIGRATIONS_STATE_PATH):
+        return {"last_runed_migration": None}
+    else:
+        try:
+            return json.load(open(MIGRATIONS_STATE_PATH))
+        except Exception as e:
+            raise MoulinetteError(
+                errno.EINVAL,
+                "Unable to load state json file located at {path}, exception: {exception)".format(
+                    exception=e,
+                    path=MIGRATIONS_STATE_PATH
+                )
+            )
+
+
+def _get_migrations_list():
+    migrations = []
+
+    try:
+        import data_migrations
+    except ImportError:
+        # not data migrations present, return empty list
+        return migrations
+
+    migrations_path = data_migrations.__path__[0]
+
+    if not os.path.exists(migrations_path):
+        logger.warn("Can't access migrations files at path %s".format(migrations_path))
+        return migrations
+
+    for migration in filter(lambda x: re.match("^\d+_[a-zA-Z0-9_]+\.py$", x), os.listdir(migrations_path)):
```

Comment:
you don't enforce the nupmber of starting digits here. is it safe to sort afterwards ?